src/fineman/finemans_algorithm.py

```diff
@@ -1,24 +1,20 @@
+import random as rand
 from collections import deque
 from math import log2
-import random as rand
-from numpy import inf
 
 from bidict import bidict
+from numpy import inf
 
 from fineman import preprocess_graph
 from fineman.dijkstra import dijkstra
 from fineman.elimination_algorithm import elimination_algorithm
-from fineman.helper_functions import reweight_graph
 
 
-def _reverse_price_functions_on_distances(source, dists, price_functions):
+def _reverse_price_functions_on_distances(source, dists, price_function):
```

Comment:
Name should be coherent with parameter or otherwise clarify better what is going on.